### PR TITLE
replace config_overrides.patch_config by config_overrides.to_ruff_args

### LINKED ISSUE #10345: Use `--config` in ecosystem checks
Our ecosystem checks manually overwrite the `pyproject.toml` file to perform some overrides but now that we have `--config` support we should use that instead.

### PR DESCRIPTION
**Pull request by hoel-bagard**
## Summary

Closes #10345

This PR removes the manual patch of the config by the overrides when doing the ecosystem checks, and instead uses the
`--config` option.

## Test Plan

CI ?

### CODE REVIEW

**Comment by zanieb** [comment]
Sorry I lost track of this. Is it ready for review?

**Comment by hoel-bagard** [comment]
It's not ready for review yet.

I'll look again for a way to unset keys. Sorry to have let this PR/issue take so long.

**Comment by hoel-bagard** [comment]
@zanieb I've tried to just set the value of the keys to unset to an empty string, but I don't know how to check whether this works or not.

When I run the `ruff-ecosystem check` locally I get a bunch of errors, but the CI is passing.

File: python/ruff-ecosystem/ruff_ecosystem/projects.py [hunk 1]
```diff
@@ -86,83 +82,21 @@ def as_string():
...
-            names = base + key.split(".")
-            for name in names[:-1]:
-                if name not in target:
-                    target[name] = {}
-                target = target[name]
-
             if value is None:
-                # Remove null values i.e. restore to default
-                target.pop(names[-1], None)
+                args.extend(["--config", f"{key} = ''"])
```
Comment (hoel-bagard) [inline python/ruff-ecosystem/ruff_ecosystem/projects.py:95]: @AlexWaygood Here I'm trying to unset / restore to default some values.

This is the part of the PR I'm the most unsure about, do you know if that's the correct way to do it ?

**Comment by MichaReiser** [comment]
I'll close this because there was no activity in the last 8 months.

**Pull request closed without merging**